setup/teardown.md

```diff
@@ -0,0 +1,73 @@
+---
+title: Teardown
+description: Learn how to teardown Coder and its associated infrastructure.
+---
+
+This guide shows you how to teardown Coder and the cluster it is deployed on.
+
+**Note:** these instructions help you remove infrastructure spun up via our
+[Kubernetes Documentation](https://coder.com/docs/setup/kubernetes), and do not
+include teardown steps for any additional resources that may have been created
+outside of these steps
```

Comment:
That being said, if there are other resources on a cluster, they will likely be deleted as well. 

Coder's recommended install path involved Coder being installed on to its own separate namespace. Perhaps the teardown docs could discuss how to delete all of the _Coder_ resources, and then separately how to destroy the cluster too?

Reply:
Good point - 

> > received the following:
> > `yarn run v1.22.10 warning package.json: No license field error Command "all-contributors" not found.`
> 
> Might mean you haven't run `yarn` to install?
> 
> I just performed the actions as a check:
> 
> ```
> greyb ~/Code/docs (vapurrmaid/chore-contributors) $ git checkout ericpaulsen/teardown 
> Branch 'ericpaulsen/teardown' set up to track remote branch 'ericpaulsen/teardown' from 'origin'.
> Switched to a new branch 'ericpaulsen/teardown'
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn
> yarn install v1.22.5
> warning package.json: No license field
> warning No license field
> [1/4] Resolving packages...
> [2/4] Fetching packages...
> [3/4] Linking dependencies...
> [4/4] Building fresh packages...
> 
> Done in 0.35s.
> greyb ~/Code/docs (ericpaulsen/teardown) $ yarn all-contributors add ericpaulsen content
> yarn run v1.22.5
> warning package.json: No license field
> $ /home/greyb/Code/docs/node_modules/.bin/all-contributors add ericpaulsen content
> Done in 0.28s.
> ```
> 
> Don't sweat it if it doesn't work - we'll have the all contributors bot up soon (thank you @bpmct). We'll make sure you get recorded!

thank you grey! I ran `git pull` to make sure I had the latest updates, for some reason I'm still receiving the error `Command "all-contributors" not found`.

Reply:
Understood - so this would involve adding steps to delete the namespace before the cluster delete commands are run. @khorne3 what are your thoughts?

Reply:
That's not a bad idea -- I think it betters match our happy path install (but in reverse obviously)

Reply:
so I've just spoken with @sreya - we actually do not have a way to delete all coder-related resources in the cluster, while keeping other unrelated resources intact (but it is possible via a script). if we were to just delete the namespace, it wouldn't remote some cluster-scoped resources associated with coder, which would actually cause a reinstall to fail.

so for now, an alternative to simply deleting the whole cluster is to run `helm uninstall coder`, which will delete the coder services but keep the user environments (and their related disk space) intact.

what are the team's thoughts? @bpmct @khorne3

Reply:
Thanks for the context. With that being said, I think that we can keep these instructions and then potentially add the script if/when is is ready.

I do think it's important to mention that deleting this teardown guide removes the whole cluster, which may potentially contain non-coder applications if the user installed extra stuff.

Reply:
Agreed - we do have language in this doc that says `This guide shows you how to tear down Coder and the cluster it is deployed on.` So that should be pretty clear to the user. 

As for the `helm uninstall` route, we can add a note saying that it will _only_ delete Coder `services` within the cluster, and all user environments will persist.

Reply:
Can we adjust the callout to the following and if anyone asks, note that anything more will require some type of script? I'm afraid of spending a lot of time on something that will need modifications more often than not

> These instructions help you remove infrastructure created when following our
[Kubernetes setup tutorials](../setup/kubernetes/index.md). They do not include
teardown steps for any additional resources that you create. If you need to keep
your cluster, you can run `helm uninstall coder`, which deletes all Coder
services but retains environments and their associated disk space.

Reply:
That works well. I'm good with the changes above.